dask/dataframe/io/parquet/utils.py

```diff
@@ -10,6 +13,86 @@
 class Engine:
     """The API necessary to provide a new Parquet reader/writer"""
 
+    @classmethod
+    def extract_filesystem(
+        cls,
+        urlpath,
+        filesystem,
+        dataset_options,
+        open_file_options,
+        storage_options,
+    ):
+        """Extract filesystem object from urlpath or user arguments
+
+        This classmethod should only be overridden for engines that need
+        to handle filesystem implementations other than ``fsspec``
+        (e.g. ``pyarrow.fs.S3FileSystem``).
+
+        Parameters
+        ----------
+        urlpath: str or List[str]
+            Source directory for data, or path(s) to individual parquet files.
+        filesystem: "fsspec" or fsspec.AbstractFileSystem
+            Filesystem backend to use. Default is "fsspec"
+        dataset_options: dict
+            Engine-specific dataset options.
+        open_file_options: dict
+            Options to be used for file-opening at read time.
+        storage_options: dict
+            Options to be passed on to the file-system backend.
+
+        Returns
+        -------
+        fs: Any
+            A global filesystem object to be used for metadata
+            processing and file-opening by the engine.
+        paths: List[str]
+            List of data-source paths.
+        dataset_options: dict
+            Engine-specific dataset options.
+        open_file_options: dict
+            Options to be used for file-opening at read time.
+        """
+
+        # Check if fs was specified as a dataset option
+        fs = dataset_options.pop("fs", "fsspec")
+        if filesystem is not None:
+            fs = filesystem
+
+        if fs in (None, "fsspec"):
```

Comment:
I don't think `fs` will ever be `None` based on `fs = dataset_options.pop("fs", "fsspec")` a few lines above

Reply:
The user is free to pass in `dataset={"fs": None, ...}`, so `None` is still a possibility.